test/unit/test_manifest_download.py

```diff
@@ -64,40 +65,88 @@ def _assert_all_files_downloaded(self):
                          for dir_path, _, files in scandir.walk('.')
                          for f in files}
         files_expected = {
-            os.path.join('.', 'manifest.tsv'),
+            os.path.join('.', os.path.basename(self.manifest_file)),
             os.path.join(self.version_dir, 'ad', '3fc1', 'ad3fc1e4898e0bce096be5151964a81929dbd2a92bd5ed56a39a8e133053831d'),
             os.path.join(self.version_dir, '8f', '3507', '8f35071eaeedd9d6f575a8b0f291daeac4c1dfdfa133b5c561232a00bf18c4b4'),
             os.path.join(self.version_dir, '8f', '3404', '8f3404db04bdede03e9128a4b48599d0ecde5b2e58ed9ce52ce84c3d54a3429c'),
         }
         self.assertEqual(files_present, files_expected)
 
+    def _assert_manifest_updated_with_paths(self):
+        output_manifest = os.path.basename(self.manifest_file)
+        self.assertTrue(os.path.isfile(output_manifest))
+        with open(output_manifest, 'r') as f:
+            output_manifest = [tuple(line.split('\t')) for line in f.read().splitlines()]
+        expected_manifest = list(zip(*self.manifest))
```

Comment:
Consider removing the transposition from the definition of `self.manifest`. I think that will eliminate one transposition.

Reply:
It will remove a transpose here, but add another one in `_write_manifest()`. I think it's simpler to store the manifest in the form it's written in.